templates/patterns/gallery/gallery.ui_patterns.yml

```diff
@@ -0,0 +1,56 @@
+gallery:
+  label: "Gallery"
+  description: "Gallery of available media."
+  fields:
+    rows:
+      type: "array"
+      label: "Rows"
+      description: "Rows of gallery items"
+      preview:
+        - - classes: "ecl-col-md-4"
+            image:
+              src: "http://placehold.it/380x185"
+              alt: "Example alt text"
+              responsive: true
+            caption: "Example image caption"
+            icon: "ecl-icon--camera"
```

Comment:
I hid the class naming and took care of it on the preprocess function of the pattern.
However, after spending lots of time trying to figure out how to implement a MediaValueObject I have decided not to do it for the moment.
At this moment the gallery pattern and its usage are not completely defined and creating a value object  without knowing what it will be required to do seems a waste.

Reply:
The classes are in a preprocess.